src/components/ChatLog.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ChatEntry from './ChatEntry';
+import './ChatLog.css';
+
+const ChatLog = (props) => {
+  const chatEntries = props.entries.map((entry) => {
+    return (
+      <ChatEntry
+        key={entry.id}
+        id={entry.id}
+        sender={entry.sender}
+        body={entry.body}
+        timeStamp={entry.timeStamp}
+        liked={entry.liked}
+        toggleLike={props.toggleLike}
+      />
+    );
+  });
+
+  return <ul>{chatEntries}</ul>;
+};
+
+// proptypes
+ChatLog.prototype = {
+  chatData: PropTypes.arrayOf(PropTypes.object).isRequired,
+  toggleLike: PropTypes.func.isRequired,
+};
```

Comment:
We can validate the nested objects within `chatData` using the `shape` method:

We can also validate the nested data within `entries` by using the `shape()` method:

```python
ChatLog.propTypes = {
  chatdata: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      sender: PropTypes.string.isRequired,
      body: PropTypes.string.isRequired,
      timeStamp: PropTypes.string.isRequired,
      liked: PropTypes.bool,
    })
  ),
  toggleLike: PropTypes.func.isRequired,
};
```